lib/execution_engine2/utils/KafkaUtils.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+"""Module to provide kafka handlers for internal logging facility."""
+
+import json
+import logging
+
+from execution_engine2.db.models.models import Status
+
+logging.basicConfig(level=logging.INFO)
+from dataclasses import dataclass
+
+from confluent_kafka import Producer
+
+KAFKA_EVENT_TYPE = "job_status_update"
+CONDOR_EVENT_TYPE = "condor_request"
+DEFAULT_TOPIC = "ee2"
+
+
+def send_kafka_update_status(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdate)
+
+
+def send_kafka_condor_update(data):
+    send_message_to_kafka(data=data, data_class=KafkaCondorCommandUpdate)
+
+
+def send_kafka_update_start(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateStartJob)
+
+
+def send_kafka_update_finish(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateFinishJob)
+
+
+def send_kafka_update_cancel(data):
+    send_message_to_kafka(data=data, data_class=KafkaStatusUpdateCancelJob)
+
+
+@dataclass
+class KafkaStatusUpdate:
+    job_id: str = None
+    previous_status: str = None
+    new_status: str = None
+    scheduler_id: int = None
+    topic: str = DEFAULT_TOPIC
+    event_type: str = KAFKA_EVENT_TYPE
+    error: bool = False
+
+    def __post_init__(self):
+        if self.job_id is None:
+            raise Exception("Need to provide a job id")
+
+        if self.new_status is None:
+            raise Exception("Need to provide a new_status")
+
+        # All state transitions need a previous status, except the created one
+        if self.previous_status is None:
+            if self.new_status != Status.created.value:
+                raise Exception("Need to provide a previous_status")
+
+        # A created job may not have been able to have been submitted to condor, so it may not have a scheduler_id
+        if self.new_status not in ["created", "terminated", "canceled"]:
```

Comment:
Alternatively this can be an enum, see here: https://docs.python.org/3/library/enum.html
Then in the field definition you can make the type `new_status: StatusEnum` instead of str